analyses/cms-open-data-ttbar/analysis.py

```diff
@@ -147,62 +147,43 @@ def define_trijet_mass(df: ROOT.RDataFrame) -> ROOT.RDataFrame:
 
     # Build trijet combinations
     df = df.Define("Trijet", "ROOT::VecOps::Combinations(Jet_pt[Jet_pt_mask],3)")
-    df = df.Define("nTrijet", "Trijet[0].size()")
 
-    # Assign four-momentums to each trijet combination
+    
+    # trijet_btag is a helpful array of bool values indicating whether or not the maximum btag value in trijet is larger than 0.5 threshold
     df = df.Define(
-        "Trijet_p4",
-        """
-        ROOT::RVec<ROOT::Math::PxPyPzMVector> Trijet_p4(nTrijet);
-        for (int i = 0; i < nTrijet; ++i)
-        {
-            int j1 = Trijet[0][i];
-            int j2 = Trijet[1][i];
-            int j3 = Trijet[2][i];
-            Trijet_p4[i] = Jet_p4[j1] + Jet_p4[j2] + Jet_p4[j3];
-        }
-        return Trijet_p4;
-        """,
+            "Trijet_btag",
+            """
+            auto Jet_btagCSVV2_masked = Jet_btagCSVV2[Jet_pt_mask];
+            auto J1_btagCSVV2 = ROOT::VecOps::Take(Jet_btagCSVV2_masked, Trijet[0]);
+            auto J2_btagCSVV2 = ROOT::VecOps::Take(Jet_btagCSVV2_masked, Trijet[1]);
+            auto J3_btagCSVV2 = ROOT::VecOps::Take(Jet_btagCSVV2_masked, Trijet[2]);
```

Comment:
Thanks to argument-dependent lookup (basically C++ looks for a function also in the namespace of its arguments) this should also work and it's easier on the eyes (same below with other invocations of `Take`, `ArgMax` and `Combinations`):

```suggestion
            auto J1_btagCSVV2 = Take(Jet_btagCSVV2_masked, Trijet[0]);
            auto J2_btagCSVV2 = Take(Jet_btagCSVV2_masked, Trijet[1]);
            auto J3_btagCSVV2 = Take(Jet_btagCSVV2_masked, Trijet[2]);
```

Reply:
Do you mean I can delete `ROOT::VecOps::` anywhere it is?

Reply:
It works, but warning messages appeared about lines 139-140 
```py
        ROOT::VecOps::Construct<ROOT::Math::PxPyPzMVector>(
            ROOT::VecOps::Construct<ROOT::Math::PtEtaPhiMVector>(
```:
input_line_85:4:13: warning: use of function template name with no prior declaration in function call with explicit template arguments is a C++20 extension [-Wc++20-extensions]
            Construct<ROOT::Math::PtEtaPhiMVector>(
            ^
input_line_85:3:9: warning: use of function template name with no prior declaration in function call with explicit template arguments is a C++20 extension [-Wc++20-extensions]
        Construct<ROOT::Math::PxPyPzMVector>(
        ^

Reply:
looks like the function template needs the fully qualified name. let's leave it there just for the `Construct` calls